auth/store.go

```diff
@@ -333,7 +334,7 @@ func (as *authStore) CheckPassword(username, password string) (uint64, error) {
 		}
 
 		if user.Options != nil && user.Options.NoPassword {
-			return 0, ErrAuthFailed
+			return 0, ErrUserIsNoPassword
```

Comment:
@mitake cool, lgtm but if you update PR, I would suggest to change the err name to `ErrNoPasswordUser` or `ErrUserWithNoPassword` if you like it. Also, I guess in the future we can a test case if that's possible. Thanks!

Reply:
Got it, I used `ErrNoPasswordUser`. And I agree with adding a test case in future as an individual PR. Sorry I couldn't take a time enough.